src/muse/constraints.py

```diff
@@ -397,9 +390,9 @@ def demand(
     search_space: xr.DataArray,
     market: xr.Dataset,
     technologies: xr.Dataset,
-    year: Optional[int] = None,
+    year: int | None = None,
```

Comment:
I think writing union types this way is a Python 3.10 thing, alas: https://docs.python.org/3/library/typing.html#typing.Union

Reply:
It's a bit weird that the Python 3.9 runners don't seem to be complaining... Maybe these bits aren't covered by tests?

Reply:
These sort of things are checked by `mypy` which we are not running in MUSE because it largely lacks of consistent typing information. Maybe at runtime these things do not matter at all?

What annoys me is that I set as target python version 3.9, so if this is a 3.10 feature, it should not have been changed :(

Reply:
Ugh, that is annoying. I guess it's a `ruff` bug? :disappointed: 

Strangely, there are plenty of other places where it's left `Optional[int]`s alone.

I suppose one option would be to do this first upgrade with `pyupgrade` itself, which presumably doesn't have this problem? I guess the `ruff` hook might then change these bits again, though. Perhaps if it's being really stubborn we should just use the `pyupgrade` pre-commit hook.

Reply:
I've just found the trick in [here](https://stackoverflow.com/a/76660598). Esentially, you can use `|` with no problem from python 3.7 as long as you do `from __future__ import annotations` which we do in all the files that `ruff` changed. 

In summary, these changes are safe to go.

Reply:
In the file where that was not changed it was because we did not have that import statement.

Reply:
Ohhh I see. Well that's a relief